### DISCUSSION TITLE
List search results in the order of relevance

### DISCUSSION THREAD

**Discussion by raindropsfromsky** [Flow-Launcher/Flow.Launcher#1220] [Ideas]
Currently, the search results seem to be listed in the order of their discovery.

Instead, results that have a better match should be listed at the top.

For example, the order of listing could be as follows-
1. Items where the whole search string matches exactly, and is found at the beginning of the name.
2. Items where the whole search string matches exactly, but is found later in the name
3. Items where the search words are found, but not in contiguous positions <br>(i.e., the words are found, but the exact phrase is not found)
4. Items where the search words are found in different orders
5. Items where only some (but not all) search words are found.

**Comment by taooceros** [comment]
We do have such function for program search. Probably you don't feel it because our index search doesn't use it.

**Reply by raindropsfromsky** [reply]
I didn't get your point: If FL already has this function, why can't the user see its effect?

Secondly, if there is a function, is it for the user to use it by using a specific search syntax?
Or is this accessible using a plugin that needs to be activated?
Or did you mean that a function is created in the code, but the search does not use it (yet)?

Thirdly, what did you mean by "program search"?
Almost all my searches are for documents and media files, but not for programs (msi/exe).
I am not a programmer, so I do not search for source/object/script files at all.
Also, most program file names are cryptic, so it is nearly impossible to search using software names.

**Reply by taooceros** [reply]
Search for documents are provided by windows index or everything, which we have no control about how to query. We can have an extra level of fuzzy search upon the it, but I suspect whether it may slow flow down. @jjw24 what do you think?

**Reply by jjw24** [reply]
If we apply our own search over it, the result list will follow the standard way of how other plugins order their results and is desirable. I think our own search is fast enough to not slow down Everything/Explorer search, but I also wouldn't know for sure until I test it.

**Reply by taooceros** [reply]
> If we apply our own search over it, the result list will follow the standard way of how other plugins order their results and is desirable. I think our own search is fast enough to not slow down Everything/Explorer search, but I also wouldn't know for sure until I test it.

well that's unclear. The everything plugin should be fine since it only returns first 100 results (which is not really reasonable since it is fast enough) sorted by the option, but the explorer plugins are somewhat returning all the results being found (so possible to return huge number of results). Generally it takes 10-100ms for the program plugin to return its results, so I suspects that may increase x2 or more for the explorer plugin.

Maybe the Rank sort option mentioned by @pc223 before may be used for explorer plugin.

**Reply by jjw24** [reply]
I think we should be able to limit how many results the windows search returns, might be via the windows search indexer query.
